src/containerapp/azext_containerapp/tests/latest/test_containerapp_env_commands.py

```diff
@@ -174,6 +176,173 @@ def test_containerapp_env_identity_user(self, resource_group):
+
+        user_identity_name1 = self.create_random_name(prefix='env-msi1', length=24)
+        identity_json = self.cmd('identity create -g {} -n {}'.format(resource_group, user_identity_name1)).get_output_in_json()
+        user_identity_id1 = identity_json["id"]
+        principal_id1 = identity_json["principalId"]
+
+        # assign secret permissions to the user assigned identity
+        self.cmd(f"az keyvault set-policy -n {key_vault_name} -g {resource_group} --object-id {principal_id1} --secret-permissions get list")
+
+        # create an environment with custom domain and user assigned identity
+        self.cmd('containerapp env create -g {} -n {} --mi-user-assigned {} --logs-destination none'.format(
+            resource_group, env_name, user_identity_id1))
+        
+        containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+        while containerapp_env["properties"]["provisioningState"].lower() == "waiting":
+            time.sleep(5)
+            containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+
+        cert = self.cmd(f"containerapp env certificate upload -g {resource_group} -n {env_name} --key-vault-url {akv_secret_url} --identity {user_identity_id1}", checks=[
+            JMESPathCheck('type', "Microsoft.App/managedEnvironments/certificates"),
+        ]).get_output_in_json()
+
+        cert_name = cert["name"]
+        cert_id = cert["id"]
+        cert_thumbprint = cert["properties"]["thumbprint"]
+        cert_location = cert["location"]
+
+        self.cmd(
+            'containerapp env certificate list -n {} -g {} -l "{}"'.format(env_name, resource_group, cert_location),
+            checks=[
+                JMESPathCheck('length(@)', 1),
+                JMESPathCheck('[0].properties.certificateKeyVaultProperties.keyVaultUrl', akv_secret_url),
+                JMESPathCheck('[0].properties.certificateKeyVaultProperties.identity', user_identity_id1),
+                JMESPathCheck('[0].properties.thumbprint', cert_thumbprint),
+                JMESPathCheck('[0].name', cert_name),
+                JMESPathCheck('[0].id', cert_id),
+            ])
+
+        tmpFile = "{}.pem".format(temp.name)
+        self.cmd(f'az keyvault secret download --vault-name {key_vault_name} -n {cert_name} -f {tmpFile}')
+        self.cmd('containerapp env certificate upload -g {} -n {} --certificate-file "{}"'.format(
+            resource_group, env_name, tmpFile), checks=[
+                JMESPathCheck('type', "Microsoft.App/managedEnvironments/certificates"),
+                JMESPathCheck('properties.certificateKeyVaultProperties', None),
+            ])
+
+        # update env certificate using msi
+        self.cmd(f"containerapp env certificate upload -g {resource_group} -n {env_name} --key-vault-url {akv_secret_url} --identity {user_identity_id1}", checks=[
```

Comment:
Can we use `containerapp env certificate list` to get the certificate and its properties?